bundle/phases/plan.go

```diff
@@ -46,3 +50,32 @@ func deployPrepare(ctx context.Context, b *bundle.Bundle) map[string][]libraries
 
 	return libs
 }
+
+// checkForPreventDestroy checks if the resource has lifecycle.prevent_destroy set, but the plan calls for this resource to be recreated or destroyed.
+// If it does, it returns an error.
+func checkForPreventDestroy(b *bundle.Bundle, actions []deployplan.Action) error {
+	root := b.Config.Value()
+	var errs []error
+	for _, action := range actions {
+		if action.ActionType != deployplan.ActionTypeRecreate && action.ActionType != deployplan.ActionTypeDelete {
+			continue
+		}
+
+		path := dyn.NewPath(dyn.Key("resources"), dyn.Key(action.Group), dyn.Key(action.Key), dyn.Key("lifecycle"), dyn.Key("prevent_destroy"))
+		// If there is no prevent_destroy, skip
+		preventDestroyV, err := dyn.GetByPath(root, path)
+		if err != nil {
+			return nil
```

Comment:
Should this be a return or continue? Looks like we going to miss actions that come afterwards?